rdp.c

```diff
@@ -90,6 +90,37 @@ uint16 g_session_height;
 
 static void rdp_out_unistr(STREAM s, char *string, int len);
 
+/* reads a TS_SHARECONTROLHEADER from stream, returns True of there is
+   a PDU available otherwise False */
+static RD_BOOL
+rdp_ts_in_share_control_header(STREAM s, uint8 *type, uint16 *length)
+{
+	uint16 pdu_type;
+	uint16 pdu_source;
```

Comment:
> rdp.c:99:9: warning: variable ‘pdu_source’ set but not used [-Wunused-but-set-variable]

I get compiler warnings about this variable. Do we need the information in `pdu_source` for anything? Should we just skip reading it?

Reply:
Ah ok, sometimes we have skipped those bytes, and sometimes we have tagged the variable like `UNUSED(pdu_source)`. I like the latter best due to clean readable code for packet parsing instead of `in_uint8s(s,2); ` which just skip over bytes

Reply:
The updated patch doesn't look like it solved the "problem" - we're using `pdu_type` to set `*type` even though it's marked as unused. That doesn't make much sense to me.

We're also still writing a value to `pdu_source` but never using it for anything, even though it's marked as unused. The compiler is somehow more happy with this version, but I think it looks odd. I agree that just skipping the bytes is less than ideal.

Reply:
>We're also still writing a value to pdu_source but never using it for anything, even though it's 
> marked as unused. The compiler is somehow more happy with this version, but I think it looks 
> odd. I agree that just skipping the bytes is less than ideal.
>

I there a third option for solution ? As I see it we have a choice between , read value to tagged unused variable or use skip byte function, what do you prefer, as mentioned earlier I prefer the method of tagging unused variables over skip x bytes for clarity.